ethers-providers/src/stream/watcher.rs

```diff
@@ -0,0 +1,117 @@
+#![allow(clippy::return_self_not_must_use)]
```

Comment:
I think this is useless?

Reply:
it was already there, so i left it without thinking about it. i'll check if removing it is reasonable